shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/NearbyStaticData.kt

```diff
@@ -395,6 +382,19 @@ data class NearbyStaticData(val data: List<TransitWithStops>) {
             )
         }
 
+        private fun filterStopsByPatterns(
+            routePatterns: List<RoutePattern>,
+            global: GlobalResponse,
+            localStops: Set<String>
+        ): Set<String> {
+            val patternsStops =
+                routePatterns.flatMapTo(mutableSetOf()) {
```

Comment:
Haven't seen `flatMapTo`, very clean!

Reply:
I wish sometimes that Kotlin iterable methods were lazy like Rust iterator methods are, so that `.flatMap().toSet()` wouldn't construct the full list with duplicates, but since they aren't and it would, I thought it made sense to keep this slightly more efficient.